packages/fxa-payments-server/src/lib/amplitude.ts

```diff
@@ -0,0 +1,267 @@
+  safeLogAmplitudeEvent(
+    eventGroupNames.updatePayment,
+    eventTypeNames.engage,
+    eventProperties
+  );
+}
+
+export function updatePayment_PENDING(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.updatePayment,
+    eventTypeNames.submit,
+    eventProperties
+  );
+}
+
+export function updatePayment_FULFILLED(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.updatePayment,
+    eventTypeNames.success,
+    eventProperties
+  );
+  safeLogAmplitudeEvent(
+    eventGroupNames.updatePayment,
+    eventTypeNames.complete,
+    eventProperties
+  );
+}
+
+export function updatePayment_REJECTED(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.updatePayment,
+    eventTypeNames.fail,
+    eventProperties
+  );
+}
+
+export function cancelSubscriptionMounted(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.cancelSubscription,
+    eventTypeNames.view,
+    eventProperties
+  );
+}
+
+export function cancelSubscriptionEngaged(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.cancelSubscription,
+    eventTypeNames.engage,
+    eventProperties
+  );
+}
+
+export function cancelSubscription_PENDING(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.cancelSubscription,
+    eventTypeNames.submit,
+    eventProperties
+  );
+}
+
+export function cancelSubscription_FULFILLED(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.cancelSubscription,
+    eventTypeNames.success,
+    eventProperties
+  );
+  safeLogAmplitudeEvent(
+    eventGroupNames.cancelSubscription,
+    eventTypeNames.complete,
+    eventProperties
+  );
+}
+
+export function cancelSubscription_REJECTED(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.cancelSubscription,
+    eventTypeNames.fail,
+    eventProperties
+  );
+}
+
+export function updateSubscriptionPlanMounted(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.upgradeSubscription,
+    eventTypeNames.view,
+    eventProperties
+  );
+}
+
+export function updateSubscriptionPlanEngaged(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.upgradeSubscription,
+    eventTypeNames.engage,
+    eventProperties
+  );
+}
+
+export function updateSubscriptionPlan_PENDING(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.upgradeSubscription,
+    eventTypeNames.submit,
+    eventProperties
+  );
+}
+
+export function updateSubscriptionPlan_FULFILLED(eventProperties: EventProperties) {
+  safeLogAmplitudeEvent(
+    eventGroupNames.upgradeSubscription,
+    eventTypeNames.success,
+    eventProperties
+  );
+  safeLogAmplitudeEvent(
+    eventGroupNames.upgradeSubscription,
+    eventTypeNames.complete,
+    eventProperties
+  );
```

Comment:
It's inconsistent, but the complete event is not in the list from https://github.com/mozilla/fxa/issues/3274#issuecomment-563322095

Reply:
Cool, I can snip that out. Also noticed there are a few other things that don't have pings at all (e.g. reactivateSubscription) so we might want to review this stuff overall, next year.